app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,28 @@
+class VotesController < ApplicationController
+
+  def new
+    @vote = Vote.all
+  end
+
+  def create
+    user_id = session[:logged_in_user]
+    work_id = params[:work_id]
+    @vote = Vote.new(user_id: user_id, work_id: work_id)
+
+    if @vote.save
+      flash[:success] = "Successfully upvoted!"
+    else
+      if @vote.user_id == nil
+        flash[:failure] = "You must log in to do that!"
+      else
+        flash[:failure] = "Could not upvote."
```

Comment:
You should be more specific about what happened, like, "you cannot vote twice for the same work."